src/storage/storageController.ts

```diff
@@ -24,52 +26,50 @@ export enum DefaultSavingLocations {
   'Session Only' = 'Session Only'
 }
 
-// A saved connection does not contain the connection information,
-// just metadata about the connection and an id for referencing secure storage.
-export type SavedConnection = {
-  id: string; // uuidv4
-  name: string; // Possibly user given name, not unique.
-  storageLocation: StorageScope;
+export type ConnectionsFromStorage = {
+  [key: string]: StoreConnectionInfo
 };
 
-type StoredConnectionsType = { [key: string]: SavedConnection } | undefined;
+type StoredVariableName = StorageVariables.GLOBAL_USER_ID |
+  StorageVariables.GLOBAL_SAVED_CONNECTIONS |
+  StorageVariables.WORKSPACE_SAVED_CONNECTIONS |
+  StorageVariables.GLOBAL_HAS_BEEN_SHOWN_INITIAL_VIEW;
+
+type StoredItem<T> =
+    T extends StorageVariables.GLOBAL_USER_ID ? string :
+    T extends StorageVariables.GLOBAL_HAS_BEEN_SHOWN_INITIAL_VIEW ? boolean :
+    T extends StorageVariables.GLOBAL_SAVED_CONNECTIONS ? ConnectionsFromStorage :
+    T extends StorageVariables.WORKSPACE_SAVED_CONNECTIONS ? ConnectionsFromStorage :
+    never;
```

Comment:
(nit): How about this? It feels a bit more TypeScript-y, and it avoids a bit of duplication

```suggestion
interface StorageVariableContents {
  [StorageVariables.GLOBAL_USER_ID]: string;
  [StorageVariables.GLOBAL_HAS_BEEN_SHOWN_INITIAL_VIEW]: boolean;
  [StorageVariables.GLOBAL_SAVED_CONNECTIONS]: ConnectionsFromStorage;
  [StorageVariables.WORKSPACE_SAVED_CONNECTIONS]: ConnectionsFromStorage;
}

type StoredVariableName = keyof StorageVariableContents;
type StoredItem<T extends StoredVariableName> = StorageVariableContents[T];
```

(and as you might be able to imagine, I think it would be even simpler if instead of `StorageVariables.GLOBAL_USER_ID` we used `'GLOBAL_USER_ID'` as a constant string, but I don’t want to add too much stuff to this PR that isn’t already being touched anyway)

Reply:
Love the suggestion with the interface format! ❤️ I felt like my solution could be optimized but didn't figure out your solution by myself, thx!